src/utils/users.ts

```diff
@@ -0,0 +1,70 @@
+import { User } from "../models/user";
+
+type LocalStorageKey = "users";
+
+export const usersKey: LocalStorageKey = "users";
+
+// the index is a username and the value is a password
+type UsersObject = { [index: string]: string };
+
+// A safe alternative to localStorage.getItem which should always be used
+// instead of that function
+export const getLocalStorageItem = (key: LocalStorageKey): string | null => {
+  return window.localStorage.getItem(key);
+};
+
+export const setLocalStorageItem = (
+  key: LocalStorageKey,
+  value: string
+): void => {
+  return window.localStorage.setItem(key, value);
+};
+
+// Insert a new user into local storage when they sign up
+export const signupUser = (user: User): void => {
+  try {
+    let item = getLocalStorageItem(usersKey);
+    if (item) {
+      let oldUsers = JSON.parse(item);
+      let newUsers = insertUser(user, oldUsers);
+      setLocalStorageItem(usersKey, JSON.stringify(newUsers));
+    }
```

Comment:
I don't think any of these need to be `let`. We're not mutating any of the values. We should use `const` instead.